Integrations/integration-ServiceNow.yml

```diff
@@ -0,0 +1,1321 @@
+            sys_created_by:'Created by',
+            short_description:'Short description'
+        },
+        'servicenow-incident-upload-file': {
+            file_name: 'Filename',
+            download_link: 'Download link',
+            sys_id: 'System ID'
+        }
+    };
+
+    var ticketEc = function(data) {
+        return {
+                "ID":data["sys_id"],
+                "Creator":data["opened_by"]["value"],
+                "Assignee":data["assigned_to"]["value"],
+                "State":data["state"],
+                "Summary":data["short_description"]
+            };
+    }
+
+    var dataToEc = function(data) {
+        var ec = {};
+        if (Array.isArray(data)) {
+            ec.Ticket=[];
+            for (var i=0;i<data.length;i++) {
+                ec.Ticket.push(ticketEc(data[i]));
+            }
+        } else {
+            ec.Ticket = ticketEc(data);
+        }
+        return ec;
+    }
+
+    var dataToMd = function(command,data) {
+        var md;
+        var head='|';
+        var line='|';
+
+        var dict = (command in mdDict) ? mdDict[command] : mdDict['default'];
+
+        for (key in dict) {
+            head += dict[key] + '|';
+            line += '-|';
+        }
+
+        md = head + '\n' + line + '\n';
+        data = (Array.isArray(data))? data : [data];
+        for (var i=0;i<data.length;i++) {
+            md += '|';
+            for (key in dict) {
+                md +=flattenValue(data[i][key]) + '|'
+            }
+            md += '\n';
+        }
+        return md;
+    }
+
+    var SNGetIncident = function() {
+        var ticket_type = getTableName();
+        if (args.id) {
+            var path = "table/"+ticket_type+"/"+args['id'];
+        } else if (args.number) {
+            var path = "table/"+ticket_type + encodeToURLQuery({sysparm_query:"number="+args.number});
+        } else {
+            throw 'incindet-get requires either incident ID (sys_id) or incdent number';
+        }
+        res = sendRequest('GET',path);
+        if (res && res.obj && res.obj.result) {
+            var md = "### Service now incident\n";
+            md += dataToMd(command,res.obj.result);
+            var ec = {};
+            ec = dataToEc(res.obj.result);
+        } else {
+            var md = 'Cannot find incident\n';
+        }
+        return ( {'ContentsFormat': formats['json'], 'Type': entryTypes['note'], 'Contents': res.obj.result, "HumanReadable": md, "EntryContext": ec} );
+    }
+
+    var SNUpdate = function(){
+        var ticket_type = getTableName();
+        var body = getBody();
+        var path = "table/"+ticket_type+"/"+args["id"];
+        var res = sendRequest('PATCH',path,null,JSON.stringify(body));
+        var md = "### ServiceNow incident updated\n";
+        md += dataToMd(command,res.obj.result);
+        return ( {'ContentsFormat': formats['json'], 'Type': entryTypes['note'], 'Contents': res.obj.result, "HumanReadable": md} );
```

Comment:
Cleaner to write formats.json, entryTypes.note, etc. For all the below as well...